src/handleSubmit.js

```diff
@@ -32,6 +35,9 @@ const handleSubmit = (submit, props, valid, asyncValidate, fields) => {
             return submitResult
           }, submitError => {
             const error = submitError instanceof SubmissionError ? submitError.errors : undefined
+            if (!error) {
+              setTimeout(() => throw submitError)
+            }
             stopSubmit(error)
```

Comment:
Doesn't using `setTimeout()` make your error uncatchable? Why not just return it, so code paying attention to the promise can see it?